bliss/models/location_encoder.py

```diff
@@ -75,24 +75,30 @@ def __init__(
         self.mean_detections = mean_detections
         self.max_detections = max_detections
         self.n_bands = n_bands
+
+        assert tile_slen <= ptile_slen
         self.tile_slen = tile_slen
         self.ptile_slen = ptile_slen
-        border_padding = (ptile_slen - tile_slen) / 2
-        assert tile_slen <= ptile_slen
-        assert border_padding % 1 == 0, "amount of border padding should be an integer"
-        self.border_padding = int(border_padding)
 
-        # Number of variational parameters used to characterize each source in an image.
+        assert (ptile_slen - tile_slen) % 2 == 0
+        self.border_padding = (ptile_slen - tile_slen) // 2
+
+        # Number of distributional parameters used to characterize each source in an image.
         self.n_params_per_source = sum(param["dim"] for param in self.variational_params.values())
 
-        # There are self.max_detections * (self.max_detections + 1) total possible detections.
-        # For each param, for each possible number of detection d, there are d ways of assignment.
-        # NOTE: Dimensions correspond to the probabilities in ONE tile.
-        self.dim_out_all = int(
-            0.5 * self.max_detections * (self.max_detections + 1) * self.n_params_per_source
-            + 1
-            + self.max_detections,
-        )
+        # the number of total detections for all source counts: 1 + 2 + ... + self.max_detections
+        # NOTE: the numerator here is always even
+        n_total_detections = self.max_detections * (self.max_detections + 1) // 2
+
+        # most of our parameters describe individual detections
+        n_source_params = n_total_detections * self.n_params_per_source
+
+        # we also have parameters indicating the distribution of the number of detections
+        count_simplex_dim = 1 + self.max_detections
+
+        # the total number of distributional parameters per tile
+        self.dim_out_all = n_source_params + count_simplex_dim
```

Comment:
The old code for computing `self.dim_out_all` was shorter but harder to decipher.

Reply:
This is great!